localstack/services/s3/v3/storage/core.py

```
@@ -90,11 +90,17 @@ def read(self, s: int = -1) -> bytes | None:
     def seek(self, offset: int, whence: int = 0) -> int:
         pass
 
+    @property
     @abc.abstractmethod
     def checksum(self) -> Optional[str]:
```

Pylint (3.3.7)

[refactor] Either all return statements in a function should return an expression, or none of them should.

(R1710)